commons/src/main/java/fr/free/nrw/commons/caching/CacheController.java

```diff
@@ -0,0 +1,94 @@
+package fr.free.nrw.commons.caching;
+
+import android.util.Log;
+
+import com.github.varunpant.quadtree.Point;
+import com.github.varunpant.quadtree.QuadTree;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import fr.free.nrw.commons.upload.MwVolleyApi;
+
+public class CacheController {
+
+    private double x, y;
+    private QuadTree quadTree;
+    private Point[] pointsFound;
+    private double xMinus, xPlus, yMinus, yPlus;
+
+    private static final String TAG = CacheController.class.getName();
+
+    public CacheController() {
+        quadTree = new QuadTree(-180, -90, +180, +90);
+    }
+
+    public void setQtPoint(double decLongitude, double decLatitude) {
+        x = decLongitude;
+        y = decLatitude;
+        Log.d(TAG, "New QuadTree created");
+        Log.d(TAG, "X (longitude) value: " + x + ", Y (latitude) value: " + y);
+    }
+
+    public void cacheCategory() {
```

Comment:
In general, I find that methods are more reusable and readable if I declare needed outside dependencies as parameters instead of taking on the responsibility of knowing what the client wants. Additionally, requiring an operation across two method calls can also lead to problems (`setQtPoint`, `cacheCategory`). In this case, an alternative implementation might look something like:

    public void cacheCategory(double x, double y, List<String> categories) {
        double x = longitudeToDecimal(lat);
        double y = latitudeToDecimal(lng);
        cacheCategory(x, y, categories);
    }

    public void cacheCategory(double x, double y, List<String> categories) {
        quadTree.set(x, y, pointCatList);/
    }